model/task_json.go

```diff
@@ -53,6 +53,32 @@ var (
 	TaskJSONTagKey                 = bsonutil.MustHaveTag(TaskJSON{}, "Tag")
 )
 
+// Returns distinct list of all tags for given `projectId`
+// returns [{
+//   _id: <tag name>,
+//   obj: {
+//     created: <ISO dateime>,
+//     revision: <associated revision>,
+//   }
+// }, ...]
+func GetDistinctTagNames(projectId string) ([]interface{}, error) {
+	out := []interface{}{}
```

Comment:
Instead of unmarshalling the query results into an empty interface, can you create a concrete struct for the format of the results?